### PR TITLE
[DEVOPS-1814]: send health metrics to statsd monitor cluster

### PR DESCRIPTION
**Pull request by yashlyft**
cc @theatrus

### CODE REVIEW

File: src/hashring.c [hunk 1]
```diff
@@ -13,13 +13,15 @@
 struct hashring {
 	list_t backends;
 	void *alloc_data;
+	bool is_monitor_ring;
```
Comment (theatrus) [inline src/hashring.c]: Maybe make this a ring_type_t enum or something

typedef enum {
  RING_DEFAULT = 0,
  RING_MONITOR = 1,
} hashring_type_t;

File: src/json_config.c [hunk 2]
```diff
@@ -69,25 +74,29 @@ static void parse_server_list(const json_t* jshards, list_t ring) {
...
+		  type, aconfig->prefix, aconfig->suffix);
+
+	const json_t* jdshards = json_object_get(additional_config, "shard_map");
+	parse_server_list(jdshards, aconfig->ring);
+	stats_log("added %s cluster with %d servers", type, aconfig->ring->size);
+
+            if (!strcmp(type, "duplicate")) {
+                statsrelay_list_expand(config->dupl);
+                config->dupl->data[config->dupl->size - 1] = aconfig;
+            } else {
```
Comment (theatrus) [inline src/json_config.c]: Could avoid this if we pass in the config->dupl or the config->sstats

File: src/stats.c [hunk 3]
```diff
@@ -328,6 +398,43 @@ stats_server_t *stats_server_create(struct ev_loop *loop,
 					goto server_create_err;
 			}
 		}
+
+		if (config->send_self_stats) {
+			/**
+			  * Only single config for monitor section
+			  */
+			struct additional_config *stat = config->sstats->data[0];
```
Comment (theatrus) [inline src/stats.c]: Probably want to enforce only one sstats entry in the config? (I think we do this already but could lead to later surprises)

Comment (yashlyft) [inline src/stats.c]: Yup check has been added to hard fail if multiple entries are provided.

**Comment by theatrus** [comment]
:sheep:

**Pull request merged**